perllib/FixMyStreet/Script/TfL/AutoClose.pm

```diff
@@ -92,15 +92,16 @@ sub close_reports {
             { problem_id => $r->id },
             { order_by => 'confirmed' }
         );
-        my $earliest;
+        my $old_state = '';
+        my $last_change;
         while ( my $c = $comments->next ) {
-            if ( ($c->problem_state || '') ne 'action scheduled' ) {
-                $earliest = undef;
-                next;
+            my $new_state = $c->problem_state || '';
+            if ( $new_state eq 'action scheduled' && $new_state ne $old_state) {
+                $last_change = $c->confirmed;
```

Comment:
I think this will go wrong if the sequence of comment problem_states is "action scheduled", "", "action scheduled" - e.g. normal comment, inspector comment pre this PR, normal comment. Then this loop will take the last comment time as the action scheduled time, rather than the first comment as it should do.
I think it needs to retain the previous old_state if the new state is blank, rather than update it to blank.